lib/loadGltfUris.js

```diff
@@ -33,31 +25,59 @@ function loadGltfUris(gltf, basePath, callback) {
             else if (callback) {
                 callback();
             }
-        });
-    }
+        }
+    );
 
     return gltf;
 }
 
-function loadURI(basePath, extras, uri, asyncCallback) {
-    //Load the uri into the extras object based on the uri type
-    if (isDataUri(uri)) {
-        extras.source = dataUriToBuffer(uri);
-        process.nextTick(function() {
-            asyncCallback();
-        });
-    }
-    else {
-        var uriPath = path.join(basePath, uri);
+function loadURI(gltf, basePath, name, callback) {
+    var objects = gltf[name];
 
-        fs.readFile(uriPath, function (err, data) {
-            if (err) {
-                asyncCallback(err);
+    //Iterate through each object and load its uri
+    if (defined(objects)) {
+        var ids = Object.keys(objects);
+        async.each(ids, function(id, asyncCallback) {
+            var object = objects[id];
+            var uri = object.uri;
+            object.extras = defaultValue(object.extras, {});
+
+            //Load the uri into the extras object based on the uri type
+            if (isDataUri(uri)) {
+                object.extras.source = dataUriToBuffer(uri);
```

Comment:
I think it is good to just treat images as buffers for many pipelines we will build, but, later, we'll also want to decompress them (like JavaScript's `Image` object) so that, for example, we can build texture atlases.